pkg/controllers/state/cluster.go

```diff
@@ -384,6 +387,26 @@ func (c *Cluster) MarkPodSchedulingDecisions(podErrors map[*corev1.Pod]error, po
 	}
 }
 
+// UpdatePodHealthyNodePoolScheduledTime updates podHealthyNodePoolScheduledTime
+// for pods scheduled against nodePool that have NodeRegistrationHealthy=true
+func (c *Cluster) UpdatePodHealthyNodePoolScheduledTime(ctx context.Context, nodePoolName string, schedulableTime time.Time, pods ...*corev1.Pod) {
+	nodePool := &v1.NodePool{}
+	if err := c.kubeClient.Get(ctx, types.NamespacedName{Name: nodePoolName}, nodePool); err == nil {
+		for _, p := range pods {
+			nn := client.ObjectKeyFromObject(p)
+			// If the pod is scheduled to a nodePool and if the nodePool has NodeRegistrationHealthy=true
+			// then mark the time when we thought it can schedule to now.
+			if nodePool.StatusConditions().IsTrue(v1.ConditionTypeNodeRegistrationHealthy) {
+				c.podHealthyNodePoolScheduledTime.Store(nn, schedulableTime)
```

Comment:
This is going to continually overwrite values that are already stored, right? I think we need to check that the value isn't already there?